app/scripts/api.es6

```diff
@@ -1,17 +1,37 @@
+/* eslint camelcase: 0 */  // snake_case query params are not set by us
+
+import { Ok, Err } from 'results';
 import ckan from './utils/ckan';
+import { toQuery } from './utils/querystring';
+import { promiseResult } from './utils/functional';
+import warn from './utils/warn';
 
 
 const API_ROOT = '//data.takwimu.org/api';
-const resourceUrl = id => `${API_ROOT}/action/datastore_search?resource_id=${id}`;
+const resourceUrl = (id, params = {}) => promiseResult(
+  toQuery({...params, resource_id: id})
+    .orElse(err => {
+      warn(err);
+      return Err(['error.api.pre-request']);
+    })
+    .andThen(qs => Ok(`${API_ROOT}/action/datastore_search?${qs}`)));
+
 // const queryUrl = q => `${API_ROOT}/action/datastore_search_sql?sql=${q}`;
 
 
 export const getWaterpoints = () =>
-  ckan.get(resourceUrl('a94b3653-55f4-4455-9bed-42b92d5c4370'));
+  resourceUrl('a94b3653-55f4-4455-9bed-42b92d5c4370', {
+    limit: 500,
```

Comment:
because there's no way to say "all" with ckan datastore_search (we might have to switch to datastore_search_sql), and react-leaflet started having performance problems with more than this. (found a work-around, almost ready).

Also, there are a _lot_ of waterpoints. we might have some hard problems with this... we'll see.